lib/internal/options.js

```diff
@@ -11,8 +13,23 @@ function getOptionValue(option) {
   return result.value;
 }
 
+function getAllowUnauthorized() {
```

Comment:
Non-blocking nit: IMO the name isn't ideal since it also conditionally prints a warning. Unfortunately, an alternative name isn't coming to mind right away.

Reply:
We can rename later if one comes to mind